tests/unit/test_p11wrapper.cpp

```diff
@@ -0,0 +1,400 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstring>
+#include <string>
+
+#include "mococrw/p11_wrap.h"
+#include "p11_lib_mock.h"
+
+using namespace ::mococrw::p11;
+using namespace ::std::string_literals;
+
+using ::testing::_;
+using ::testing::AnyNumber;
+using ::testing::DoAll;
+using ::testing::Mock;
+using ::testing::Return;
+using ::testing::SetArgPointee;
+
+/**
+ * Test the P11 wrapper.
+ *
+ * These tests are to determine that the p11-wrapper,
+ * declared and defined in p11_wrap.h and p11_wrap.cpp
+ * behaves as intended and calls the underlying p11 library
+ * correctly.
+ *
+ */
+class P11WrapperTest : public ::testing::Test
+{
+public:
+    void SetUp() override;
+    void TearDown() override;
+
+protected:
+    LibP11Mock &_mock() const { return LibP11MockManager::getMockInterface(); }
+};
+
+namespace testutils
+{
+PKCS11_CTX *somePKCS11CtxPtr()
```

Comment:
purpose?

Reply:
Since we are mocking, this code returns a dummy pointer to a PKCS11_CTX value.